rskj-core/src/test/java/co/rsk/peg/union/UnionBridgeStorageIndexKeyTest.java

```diff
@@ -3,23 +3,39 @@
 import static org.junit.jupiter.api.Assertions.*;
 
 import org.ethereum.vm.DataWord;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
 
 class UnionBridgeStorageIndexKeyTest {
 
-    @Test
-    void getKey_unionBridgeContractAddress_shouldReturnKey() {
-        UnionBridgeStorageIndexKey key = UnionBridgeStorageIndexKey.UNION_BRIDGE_CONTRACT_ADDRESS;
-
-        DataWord expectedKey = DataWord.fromString("unionBridgeContractAddress");
-        assertEquals(expectedKey, key.getKey());
-    }
-
-    @Test
-    void getKey_unionBridgeLockingCap_shouldReturnKey() {
-        UnionBridgeStorageIndexKey key = UnionBridgeStorageIndexKey.UNION_BRIDGE_LOCKING_CAP;
-
-        DataWord expectedKey = DataWord.fromString("unionBridgeLockingCap");
-        assertEquals(expectedKey, key.getKey());
+    @ParameterizedTest
+    @EnumSource(UnionBridgeStorageIndexKey.class)
+    void getKey_shouldReturnCorrectKey(UnionBridgeStorageIndexKey key) {
+        // Testing that getKey() returns the correct DataWord for each enum value
+        switch (key) {
```

Comment:
Use the new switch provided by java 17, it's a lot cleaner

Reply:
Updated